cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaRebalanceAssemblyOperator.java

```diff
@@ -413,29 +434,29 @@ private KafkaRebalanceStatus buildRebalanceStatus(String sessionID, KafkaRebalan
     }
 
     /**
-     * This method handles the transition from {@code NotReady} state.
-     * This state indicates that the rebalance has suffered some kind of error. This could be a misconfiguration or the result
-     * of an error during a reconcile.
+     * This method handles the transition from the {@code NotReady} state.
+     * This state indicates that the rebalance has suffered some kind of error.
+     * This could be due to misconfiguration or the result of an error during a reconcile.
      *
      * @param reconciliation Reconciliation information
      * @param host Cruise Control service to which sending the rebalance proposal request
      * @param apiClient Cruise Control REST API client instance
      * @param kafkaRebalance Current {@code KafkaRebalance} resource
      * @param rebalanceAnnotation The current value for the strimzi.io/rebalance annotation
      * @param rebalanceOptionsBuilder builder for the Cruise Control REST API client options
-     * @return a Future with the next {@code KafkaRebalanceStatus} bringing the state
+     * @return a Future with the next {@code KafkaRebalanceStatus} including the state
      */
     private Future<KafkaRebalanceStatus> onNotReady(Reconciliation reconciliation,
                                                     String host, CruiseControlApi apiClient,
                                                     KafkaRebalance kafkaRebalance,
                                                     KafkaRebalanceAnnotation rebalanceAnnotation,
                                                     RebalanceOptions.RebalanceOptionsBuilder rebalanceOptionsBuilder) {
-        if (rebalanceAnnotation == KafkaRebalanceAnnotation.refresh) {
-            // the user fixed the error on the resource and want to "refresh", actually
-            // requesting a new rebalance proposal
+        if (KafkaRebalanceAnnotation.refresh.equals(rebalanceAnnotation)) {
```

Comment:
the `equals` implementation for enum is just `this == other`, tbh I think that using `equals` for enums is less readable. Furthermore, I found articles talking about preferring `==` instead of `equals` just for enums.

Reply:
@samuel-hawker do you think to address this comment?